backend/app/controllers/uoas.js

```diff
@@ -157,40 +159,115 @@ module.exports = {
-                        '/ 1000.0)) WHERE "productId" = ' + req.body.productId +
-                        'AND "uoaId" = ' + req.params.id
-                    );
+                if (_.first(project)) {
+                    console.log(`I FOUND A PROJECT: ${project[0].codeName}`);
+                    // If project is not active, delete UAO without any issues
+                    if (project[0].status === 0) {
+                        yield thunkQuery(
+                            'UPDATE "ProductUOA"' +
+                            'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                            '/ 1000.0)) WHERE "UOAid" = ' + req.params.id +
+                            'AND "productId" = ' + req.body.productId
+                        );
+
+                        // Soft delete the task with that UAO ID
+                        yield thunkQuery(
+                            'UPDATE "Tasks"' +
+                            'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                            '/ 1000.0)) WHERE "productId" = ' + req.body.productId +
+                            'AND "uoaId" = ' + req.params.id
+                        );
+
+                        yield thunkQuery(
+                            'UPDATE "UnitOfAnalysis"' +
+                            'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                            '/ 1000.0)) WHERE "id" = ' + req.params.id
+                        );
+                    } else { // Project is active and we have to do other checks.
+                        // check if there are any tasks assigned
+                        var task = yield thunkQuery(
+                            'SELECT "Tasks".* ' +
+                            'FROM "Tasks" ' +
+                            'WHERE "Tasks"."uoaId" = ' + req.params.id +
+                            'AND "Tasks"."productId" = ' + req.body.productId
+                            // 'AND "Tasks"."isComplete" is True '
+                        );
+
+                        if (!_.first(task)) {
+                            // Soft delete the UOA from the Product UAO Table
+                            yield thunkQuery(
+                                'UPDATE "ProductUOA"' +
+                                'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                '/ 1000.0)) WHERE "UOAid" = ' + req.params.id +
+                                'AND "productId" = ' + req.body.productId
+                            );
+
+                            // Soft delete from the UOA table
+                            yield thunkQuery(
+                                'UPDATE "UnitOfAnalysis"' +
+                                'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                '/ 1000.0)) WHERE "id" = ' + req.params.id
+                            );
+
+                        } else {
+                            if (task[0].isComplete === true) {
```

Comment:
I think this does not quite satisfy the condition in the paper doc. This should check not that the task is complete, but if any question has been answered for the subject. Also, we should be making the check for every task, not just the first

Reply:
The condition to check whether any question has been answered will be done in a separate ticket. 582.

Do we not care that a task is completed at all? If a task is completed, that means they have answered all questions right?